### ISSUE TITLE
DBeaver freezes before a execution select script

### ISSUE THREAD

**Issue by lucasoz** [dbeaver/dbeaver#14053]
#### System information: 
- Operating system (distribution) and version
Ubuntu Versión 20.04.3 LTS (Focal Fossa) de 64 bits
Núcleo Linux 5.4.0-88-generic x86_64
MATE 1.24.0
- DBeaver version
21.2.1
- Additional extensions

#### Connection specification:
- Database name and version
PostgreSQL 12.7
- Driver name
PostgreSQL JDBC Driver 42.2.5
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Describe the problem you're observing:
Dbeaber crash before execution of a select query script
#### Steps to reproduce, if exist:
```sql
select t1.*
  from  schema_one.table_one t1
  left join schema_one.table_two t2
    on t1.id = t2.table_one_id
    and t2.user_id != t1.owner_id
  left join schema_one.table_three t3
    on t3.table_one_id = t1.id
  left join schema_one.table_four t4
    on t3.id = t4.table_three_id
  group by t1.id
  order by case
    when max(t2."date") is null and max(t3."date") is null and max(t4."date") is null then
      0
    else (
      (
        (
          (
            2 * count(distinct t2.*)
          ) + count(distinct t3.*) + count(distinct t4.*)
        ) - 1
      ) / (
        (
          (
            extract(
              epoch from (
                now() - GREATEST(
                  max(t2."date"),
                  max(t3."date"),
                  max(t4."date")
                )
              )::interval
            ) / 3600
          ) + 2
        ) ^ 1.8
      )
    )
  end desc;
```

NOTE: When I wrap this query in a function it works perfectly both execution and parse
#### Include any warning/errors/backtraces from the logs
App crash then I has not logs to show

**Comment by LonwoLonwo** [comment]
Hello @lucasoz 
Thanks for the bug report!

I created a ticket on the @JSQLParser board. We will wait for help from them.

**Comment by manticore-projects** [comment]
Greetings.

Thank you for reporting, it relates to `Count( DISTINCT a.* )` which is fixed in https://github.com/JSQLParser/JSqlParser/pull/1350 already.

The PR has not been merged yet but you could replace the JSQLParser JAR with the attached one (including this fix) for immediate remedy. Do let me know if you need any further assistance.

[jsqlparser-4.3-SNAPSHOT.zip](https://github.com/dbeaver/dbeaver/files/7255666/jsqlparser-4.3-SNAPSHOT.zip)

**Comment by LonwoLonwo** [comment]
Steel waiting for the official 4.3 JSQLParser release.

**Comment by manticore-projects** [comment]
Greetings DBeaver Team.

unfortunately, I am not sure if a 4.3 release will happen too soon. Compared to the 4.2 release only a few fixes and improvements have appeared yet. It is all up the project owner @wumpz (although I will write him an e-mail asking for help).

Is there anything I can do for you in the meantime?
What would be your preferred alternative in case a resource starved 3rd party dependency does not update often enough? (H2 was another example.)

**Comment by LonwoLonwo** [comment]
Thanks for the answer @manticore-projects 

Okay, I understood about releases. 

Maybe you can take a look at this ticket? https://github.com/dbeaver/dbeaver/issues/14735

**Comment by wumpz** [comment]
I will look into the JSqlparser PR and if all goes well the release will be done in the next few hours. Sorry for the delay.

**Comment by wumpz** [comment]
@LonwoLonwo:  [JSQLParser/JSqlParser#1442. ](https://github.com/JSQLParser/JSqlParser/issues/1442)

**Comment by LonwoLonwo** [comment]
@wumpz Thank you very much!

**Comment by uslss** [comment]
fixed along with #15155
